File: string/reverse/reverse.c


void recursive(char* s,int i, int j)
{
    if(i <j)
    {
        *(s +i) = *(s +i) + *(s +j);
        *(s +j) = *(s +i) - *(s +j);
        *(s +i) = *(s +i) - *(s +j);
        recursive(s,i +1,j -1);
    }
}


void iterative(char* s, int sSize)
{
    int i = 0;
    int j = sSize -1;

    while(i <j)
    {
        *(s +i) += *(s +j);
        *(s +j)  = *(s +i) - *(s +j);
        *(s +i) -= *(s +j);
        ++i;
        --j;
    }
}

File: string/reverse/reverse.js


var recursive = function(s,i,j) {

    if(j == null) j = s.length -1;
    if(i == null) i = 0;
    if(i <j) {

        [ s[i],s[j] ] = [ s[j],s[i] ];
        recursive(s,i +1,j -1)
    }
}


var iterative = function(s) {

    var i = 0;
    var j = s.length -1;

    while(i <j) {[ s[i],s[j] ] = [ s[j],s[i] ]; --j; ++i }
}

